picasso/src/main/java/com/squareup/picasso3/PlatformLruCache.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.picasso3
+
+import android.graphics.Bitmap
+import android.util.LruCache
+import androidx.core.graphics.BitmapCompat
+
+typealias Size = Int
+
+/** A memory cache which uses a least-recently used eviction policy.  */
+internal class PlatformLruCache(maxByteCount: Int) {
+
+  /** Create a cache with a given maximum size in bytes.  */
+  val cache =
+    object : LruCache<String, Pair<Bitmap, Size>>(if (maxByteCount != 0) maxByteCount else 1) {
```

Comment:
Use of a `Pair` causes boxing. We're already allocating the `Pair`, might as well just go back to allocating our own type which won't box the size.

Reply:
Makes sense.  Updated.